### ISSUE TITLE
Internal compiler error with `2021-03-09` in veloren

### ISSUE THREAD

**Issue by xMAC94x** [rust-lang/rust-clippy#6890]
Hey, we noticed an internal compiler error in veloren source code when updating from nightly-2021-01-01 to nightly-2021-03-09

```rust
  Compiling veloren-common v0.8.0 (/mnt/nfs/marcel/Entw/Rust/veloren/common)
error: internal compiler error: compiler/rustc_typeck/src/collect/type_of.rs:278:17: associated type missing default
   --> common/src/vol.rs:167:5
    |
167 |     type IntoIter: Iterator<Item = Vec3<i32>>;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

thread 'rustc' panicked at 'Box<Any>', /rustc/8f349be27815d43d462a32faeb270a22a68486b6/library/std/src/panic.rs:59:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: Clippy version: clippy 0.1.52 (8f349be 2021-03-08)

query stack during panic:
#0 [type_of] computing type of `vol::IntoPosIterator::IntoIter`
#1 [analysis] running analysis passes on this crate
end of query stack
```

steps to reproduce:
1. download veloren; 9612592fac948b2390a13facb586c4a22216dd02
2. adjust rust-toolchain file to `2021-03-09`
3. run clippy: cargo clippy --all-targets --locked -- -D warnings

it will throw many new warnings and error, but eventually also throw the above error.

Keep in mind that for vek::Vec we are using some weird SIMD behavior which already caused problems in the past, idk if this relates to clippy though. hope you can help :)

**Comment by Y-Nak** [comment]
This seems duplicates of #6882 and this has been already fixed in #6812.

**Comment by matthiaskrgr** [comment]
Yeah I also think its already fixed.
I was not able to reproduce the crash locally with the latest version of clippy and veloren @ bf3e12f39a4

**Comment by xMAC94x** [comment]
> Yeah I also think its already fixed.
> I was not able to reproduce the crash locally with the latest version of clippy and veloren @ bf3e12f39a4

**Comment by matthiaskrgr** [comment]
Clippy was synced 2 days ago, I think the latest night should include the fix.

**Comment by Y-Nak** [comment]
I think Clippy is not synced yet, though the PR is opened. Or my understanding about the sync process is wrong.

**Comment by matthiaskrgr** [comment]
Should be fixed "officially" with `rustc 1.52.0-nightly (acca81892 2021-03-13)`
